src/components/File.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react'
+import classNames from 'classnames'
+import { Link } from 'react-router'
+
+import styles from '../styles/table'
+import { translate } from '../lib/I18n'
+import FilenameInput from '../components/FilenameInput'
+
+const splitFilename = filename => {
+  let dotIdx = filename.lastIndexOf('.') - 1 >>> 0
```

Comment:
Pure detail but I am not a big fan of this trick, I find it not enough self-explicit. What @CPatchane thinks about it ?